Formula/autorest.rb

```diff
@@ -0,0 +1,27 @@
+class Autorest < Formula
+  desc "Swagger (OpenAPI) Specification code generator"
+  homepage "https://github.com/Azure/autorest"
+  url "https://github.com/Azure/autorest/releases/download/AutoRest-0.16.0/autorest.0.16.0.zip"
+  sha256 "bc909aaf3db2105a5d709c940c4d120c5b71047539c2068d484ced1be47650c3"
+
+  depends_on "mono"
+
+  resource "swagger" do
+    url "https://raw.githubusercontent.com/Azure/autorest/master/Samples/petstore/petstore.json"
+    sha256 "8de4043eff83c71d49f80726154ca3935548bd974d915a6a9b6aa86da8b1c87c"
+  end
+
+  def install
```

Comment:
100 minutes isn't long by our standards - `gcc` and `qt` both take substantially longer on average. That's part of the reason we provide bottles -- they give us all of the control of source builds without any of the time issues.

I'm personally in favor of keeping this in a [custom tap](https://github.com/Homebrew/brew/blob/master/docs/How-to-Create-and-Maintain-a-Tap.md) until upstream makes their build procedure more clear/practical on OS X.